components/script/dom/cookiestore.rs

```diff
@@ -0,0 +1,568 @@
+            Cow::Owned(options.value.to_string()),
+        ));
+        // TODO: This currently doesn't implement all the "set a cookie" steps which involves
+        // additional processing of the name and value
+
+        self.in_flight.borrow_mut().push_back(p.clone());
+
+        // 6. Run the following steps in parallel:
+        let _ = self
+            .global()
+            .resource_threads()
+            .send(CoreResourceMsg::SetCookieForUrlAsync(
+                self.store_id,
+                creation_url.clone(),
+                Serde(cookie.build()),
+                NonHTTP,
+            ));
+
+        // 7. Return p
+        p
+    }
+
+    /// <https://cookiestore.spec.whatwg.org/#dom-cookiestore-delete>
+    fn Delete(&self, name: USVString, can_gc: CanGc) -> Rc<Promise> {
+        // 1. Let settings be this’s relevant settings object.
+        let global = self.global();
+
+        // 2. Let origin be settings’s origin.
+        let origin = global.origin();
+
+        // 5. Let p be a new promise.
+        let p = Promise::new(&global, can_gc);
+
+        // 3. If origin is an opaque origin, then return a promise rejected with a "SecurityError" DOMException.
+        if !origin.is_tuple() {
+            p.reject_error(Error::Security, can_gc);
+            return p;
+        }
+
+        self.in_flight.borrow_mut().push_back(p.clone());
+
+        // 6. Run the following steps in parallel:
+        // TODO: the spec passes additional parameters to _delete a cookie_ that we don't handle yet
+        let _ = global
+            .resource_threads()
+            .send(CoreResourceMsg::DeleteCookieAsync(
+                self.store_id,
+                global.creation_url().clone(),
+                name.0,
+            ));
+
+        // 7. Return p.
+        p
+    }
+
+    /// <https://cookiestore.spec.whatwg.org/#dom-cookiestore-delete-options>
+    fn Delete_(&self, options: &CookieStoreDeleteOptions, can_gc: CanGc) -> Rc<Promise> {
+        // 1. Let settings be this’s relevant settings object.
+        let global = self.global();
+
+        // 2. Let origin be settings’s origin.
+        let origin = global.origin();
+
+        // 5. Let p be a new promise.
+        let p = Promise::new(&global, can_gc);
+
+        // 3. If origin is an opaque origin, then return a promise rejected with a "SecurityError" DOMException.
+        if !origin.is_tuple() {
+            p.reject_error(Error::Security, can_gc);
+            return p;
+        }
+
+        self.in_flight.borrow_mut().push_back(p.clone());
+
+        // 6. Run the following steps in parallel:
+        // TODO: the spec passes additional parameters to _delete a cookie_ that we don't handle yet
+        let _ = global
+            .resource_threads()
+            .send(CoreResourceMsg::DeleteCookieAsync(
+                self.store_id,
+                global.creation_url().clone(),
+                options.name.to_string(),
+            ));
```

Comment:
Here and elsewhere you could log an error if the send fails. In theory it never should.